model/task/task.go

```diff
@@ -1436,12 +1405,57 @@ func FindMidwayTaskFromIds(t1Id, t2Id string) (*Task, error) {
 	}
 	t2, err := FindOneId(t2Id)
 	if err != nil {
-		return nil, errors.Wrapf(err, "finding task id %s", t2Id)
+		return nil, errors.Wrapf(err, "finding task id '%s'", t2Id)
 	}
 	if t2 == nil {
 		return nil, errors.Errorf("could not find task id '%s'", t2Id)
 	}
-	return findMidwayTask(*t1, *t2)
+
+	// The tasks should be the same build variant, display name, project, and requester.
+	catcher := grip.NewBasicCatcher() // Makes an error accumulator
+	catcher.ErrorfWhen(t1.BuildVariant != t2.BuildVariant, "given tasks have differing build variants '%s' and '%s'", t1.BuildVariant, t2.BuildVariant)
+	catcher.ErrorfWhen(t1.DisplayName != t2.DisplayName, "given tasks have differing display name '%s' and '%s'", t1.DisplayName, t2.DisplayName)
+	catcher.ErrorfWhen(t1.Project != t2.Project, "given tasks have differing project '%s' and '%s'", t1.Project, t2.Project)
+	catcher.ErrorfWhen(t1.Requester != t2.Requester, "given tasks have differing requester '%s' and '%s'", t1.Requester, t2.Requester)
+	if catcher.HasErrors() {
+		return nil, catcher.Resolve()
+	}
+
+	middleOrderNumber := (t1.RevisionOrderNumber + t2.RevisionOrderNumber) / 2
+	filter, sort := ByBeforeRevision(middleOrderNumber+1, t1.BuildVariant, t1.DisplayName, t1.Project, t1.Requester)
+	query := db.Query(filter).Sort(sort)
+
+	task, err := FindOne(query)
+	if err != nil {
+		return nil, errors.Wrapf(err, "finding task between '%s' and '%s'", t1Id, t2Id)
+	}
+	if task == nil {
+		return nil, errors.Errorf("could not find task between '%s' and '%s'", t1Id, t2Id)
+	}
+
+	lowerBoundTask := t1
+	upperBoundTask := t2
+	// If t1 is after t2, t1 is our upper bound and t2 is our lower bound.
+	if t1.RevisionOrderNumber > t2.RevisionOrderNumber {
+		upperBoundTask = t1
+		lowerBoundTask = t2
+	}
+	if task.RevisionOrderNumber >= upperBoundTask.RevisionOrderNumber ||
```

Comment:
Shouldn't we stop running stepback if this occurs? This seems to be an invalid situation

Reply:
We stop stepback when we hith there, [here](https://github.com/evergreen-ci/evergreen/pull/7785/files#diff-4bef865c41d6d16e1056e61806c23fb8bbc896b3739472dd60f0dd463ef5d2b9R620). To be less than or greater than the bounds is invalid, I don't expect it ever to be called, this is more of a fallback so a stepback doesn't go rouge somehow.